### ISSUE TITLE
Firefox memory leak

### ISSUE THREAD

**Issue by vrobertson** [x0a/uBO-YouTube#13]
The "YouTube Channel Whitelist for uBlock Origin" Firefox extension possibly causes a memory leak.

Preconditions:
Firefox version 76.0.1 64 bit on Windows 7 64 bit.
Extension version 0.10.3
All installed extensions besides uBO-YouTube are disabled.
The extension settings:
6 items in "Whitelisted Channels"
All "Ad Options" on Ads tab are off.

Issue:
During playback of long video on YouTube I see the slow increase of memory amount allocated by Firefox process. It grows up to several gigabytes.

Using about:memory I've recorded some memory reports and here is the result (see a screenshot).
![jeMVbmxQKG](https://user-images.githubusercontent.com/43160208/82084846-34417080-9716-11ea-8f00-724441823218.png)
This is "Load and diff" result of the comparison of two snapshots, the first is at the start of playback and the second is after 40 minutes of playback on x2 speed.

The allocated memory released only after YouTube tab closing.

When the uBO-YouTube extension is disabled, the memory consumption doesn't grow.

How can I help you to diagnose the issue?

Thank You.

**Comment by x0a** [comment]
Hey, thanks for the detailed report. I'll start some debugging now but it might take a while to pinpoint exactly what's going on because this extension does not allocate any ArrayBuffers in the actual content/injection scripts, only in the options pages. 

So YouTube might be reacting to some of the data we are modifying on the page. I'll start by eliminating modifications one by one until we figure out what YouTube is reacting to

**Comment by x0a** [comment]
So through a tedious testing process (two instances of FF using `firefox -no-remote -ProfileManager`, playing the videos simultaneously with one browser running uBO+uBO-yt and the other just uBO), I was able to both confirm the bug was caused by our extension and what was causing it.

It seems that the culprit is here https://github.com/x0a/uBO-YouTube/blob/5cacd9f215668a108dd9f6eefed0a116f5f012b5/src/inject/index.ts#L1431-L1436

It was added to prevent YT from detecting that the page was clicked away from, and allow me to debug JS event listeners in the future. However it seems that keeping track of events was preventing Firefox from garbage collecting some of those ArrayBuffers.

Simple fix was to only track event listeners if a debug argument is passed. 

I'll push an update tonight and it should be available in an hour.
[edit] v.0.10.4 is now live on Mozilla Addons, goto Options -> Add-ons ->(the Settings Cog) -> Check for updates. Let me know if the fix worked, if not, re-open the issue

Thanks for bringing this to my attention!
Feel free to open another issue if you notice any other issues
